src/switch_ivr_async.c

```diff
@@ -1280,14 +1281,67 @@ static void send_record_stop_event(switch_channel_t *channel, switch_codec_imple
 			snprintf(buffer_name, sizeof(buffer_name), "record_url_%d", record_index);
 			switch_channel_set_variable_printf(channel, buffer_name, "%s", !zstr(rh->file) ? rh->file : "");
 
-			if ((!zstr(prev_record_sec_str) && switch_is_number(prev_record_sec_str))
-				&& !zstr(prev_record_ms_str) && switch_is_number(prev_record_ms_str)) {
-				updated_record_seconds += atoi(prev_record_sec_str);
-				updated_record_ms += atoi(prev_record_ms_str);
-			}
+			/* Handle labeled CDR variables if record_label is set */
+			if (!zstr(rh->record_label)) {
+				/* For labeled recordings, accumulate in label-specific variables */
+				switch_size_t labeled_record_seconds = current_record_seconds;
+				switch_size_t labeled_record_ms = current_record_ms;
+				const char *prev_labeled_sec_str;
+				const char *prev_labeled_ms_str;
+				int prev_sec;
+				int prev_ms;
+				
+				/* Get previous labeled seconds and ms if they exist */
+				snprintf(buffer_name, sizeof(buffer_name), "record_%s_seconds", rh->record_label);
+				prev_labeled_sec_str = switch_channel_get_variable(channel, buffer_name);
+				
+				snprintf(buffer_name, sizeof(buffer_name), "record_%s_ms", rh->record_label);
+				prev_labeled_ms_str = switch_channel_get_variable(channel, buffer_name);
+				
+				/* Add to previous values if both exist (like original code) */
+				if ((!zstr(prev_labeled_sec_str) && switch_is_number(prev_labeled_sec_str))
+					&& (!zstr(prev_labeled_ms_str) && switch_is_number(prev_labeled_ms_str))) {
+					prev_sec = switch_safe_atoi(prev_labeled_sec_str, 0);
+					prev_ms = switch_safe_atoi(prev_labeled_ms_str, 0);
+					if (prev_sec >= 0 && prev_ms >= 0) {
+						labeled_record_seconds += (switch_size_t)prev_sec;
+						labeled_record_ms += (switch_size_t)prev_ms;
+					}
+				}
+				
+				/* Set the labeled CDR variables */
+				snprintf(buffer_name, sizeof(buffer_name), "record_%s_seconds", rh->record_label);
+				switch_channel_set_variable_printf(channel, buffer_name, "%d", labeled_record_seconds);
+				
+				snprintf(buffer_name, sizeof(buffer_name), "record_%s_ms", rh->record_label);
+				switch_channel_set_variable_printf(channel, buffer_name, "%d", labeled_record_ms);
+				
+				switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_DEBUG,
+					"Recording stopped with label '%s': current=%ds/%dms, total=%ds/%dms\n",
+					rh->record_label, 
+					(int)current_record_seconds, (int)current_record_ms,
+					(int)labeled_record_seconds, (int)labeled_record_ms);
+			} else {
+				/* For non-labeled recordings, accumulate in record_seconds and record_ms */
+				/* Use same logic as original code - both variables must exist */
+				if ((!zstr(prev_record_sec_str) && switch_is_number(prev_record_sec_str))
+					&& (!zstr(prev_record_ms_str) && switch_is_number(prev_record_ms_str))) {
+					int prev_sec = switch_safe_atoi(prev_record_sec_str, 0);
+					int prev_ms = switch_safe_atoi(prev_record_ms_str, 0);
+					if (prev_sec >= 0 && prev_ms >= 0) {
```

Comment:
what will happen if those are negative or equal to zero? perhaps to add an error log and return an err?